changelogs/unreleased/gh-9965-wrong-numbers-comparison.md

```diff
@@ -0,0 +1,30 @@
+## bugfix/core
+
+- Fixed several bugs in the numbers comparison and hashing in tuple keys. It
+  allowed to insert the same key multiple times into a unique index, and
+  sometimes wouldn't allow to find an existing key in an index. Could happen
+  when numbers were encoded in MessagePack suboptimally and when `double` field
```

Comment:
```suggestion
  when numbers were encoded in MessagePack suboptimally and when the `double` field
```

Reply:
I've pushed an update. To master as well. Resolving this since the PR is approved. Same for all the other comments.